### PR TITLE
ci: docker builds set their build version

### PR DESCRIPTION
**Pull request by gord5500**
### Why

We forgot to do this for the docker builds

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -105,19 +105,19 @@ jobs:
...
+          RELEASE_ID: ${{ needs.create_release.outputs.release_id }}
           PLATFORM: ${{ matrix.platform }}
           FILE_NAME: ${{ matrix.file_name }}
           GH_TOKEN: ${{ github.token }}
         working-directory: src/D2L.Bmx/build
         run: |
           if ($env:PLATFORM -eq 'win') {
             Compress-Archive -Path "bmx.exe" -DestinationPath "bmx-win-x64.zip"
-            gh release upload "$env:RELEASE_TAG" "bmx-win-x64.zip"
+            gh release upload "$env:RELEASE_ID" "bmx-win.zip"
```
Comment (cfbao) [inline .github/workflows/publish.yml:116]: please sort out all the `-x64` changes

File: Dockerfile.alpine [hunk 2]
```diff
@@ -1,5 +1,8 @@
 FROM amd64/alpine:3.15 AS build
 
+ARG version
+ARG information_version
```
Comment (cfbao) [inline Dockerfile.alpine:4]: Thinking of local builds - what happens when we don't provide these arguments on the command line when invoking docker build? Should we set default values for them?

File: Dockerfile.alpine [hunk 3]
```diff
@@ -25,7 +28,13 @@ COPY src/D2L.Bmx/D2L.Bmx.csproj src/D2L.Bmx/D2L.Bmx.csproj
 RUN dotnet restore src/D2L.Bmx -r linux-musl-x64
 
 COPY . .
-RUN dotnet publish src/D2L.Bmx --no-restore -r linux-musl-x64 -o app
+RUN dotnet publish src/D2L.Bmx \
+    -p:Version=$version \
+    -p:InformationVersion=$information_version \
```
Comment (cfbao) [inline Dockerfile.alpine:33]: variable references should be quoted in shell scripts as a best practice
```suggestion
    -p:Version="$version" \
    -p:InformationVersion="$information_version" \
```

File: Dockerfile.alpine (hunk 2 above)
Comment (gord5500) [reply Dockerfile.alpine:4]: Yea I can set them to default to 3.0.0 and v3.0.0

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -152,23 +162,23 @@ jobs:
...
-          RELEASE_TAG: ${{ needs.create_release.outputs.release_id }}
+          RELEASE_ID: ${{ needs.create_release.outputs.release_id }}
           PLATFORM: ${{ matrix.platform }}
           GH_TOKEN: ${{ github.token }}
         working-directory: build
         run: |
           chmod +x "bmx"
-          tar -czvf "bmx-$PLATFORM-x64.tar.gz" "bmx"
-          gh release upload "$RELEASE_TAG" "bmx-$PLATFORM-x64.tar.gz"
+          tar -czvf "bmx-$PLATFORM.tar.gz" "bmx"
```
Comment (cfbao) [inline .github/workflows/publish.yml:171]:
```suggestion
          tar -czvf "bmx-$PLATFORM-x64.tar.gz" "bmx"
```

Comment (gord5500) [reply .github/workflows/publish.yml:171]: A sorry 😢

**Review by cfbao — APPROVED** [review]

**Pull request merged**